packages/astro/src/core/add/index.ts

```diff
@@ -727,6 +739,128 @@ export async function validateIntegrations(integrations: string[]): Promise<Inte
 	}
 }
 
+async function updateTSConfig(
+	cwd = process.cwd(),
+	logging: LogOptions,
+	integrationsInfo: IntegrationInfo[],
+	flags: yargs.Arguments
+): Promise<UpdateResult> {
+	const frameworkWithTSSettings = ['solid-js', 'preact', 'react', 'vue'] as const;
+	const integrations = integrationsInfo.map(
+		(integration) => integration.id as 'solid-js' | 'preact' | 'react' | 'vue'
+	);
+	const firstIntegrationWithTSSettings = integrations.find((integration) =>
+		frameworkWithTSSettings.includes(integration)
+	);
+
+	if (!firstIntegrationWithTSSettings) {
+		return UpdateResult.none;
+	}
+
+	let inputConfig = loadTSConfig(cwd);
```

Comment:
# Loading needs

- Load the tsconfig / jsconfig
- Know if it exists
- Get a config object
- With the comments intact?
  - can't merge so maybe not